src/main/groovy/com/cloudbees/groovy/cps/CpsTransformer.groovy

```diff
@@ -78,7 +79,7 @@ import static org.codehaus.groovy.syntax.Types.*
  * @author Kohsuke Kawaguchi
  */
 class CpsTransformer extends CompilationCustomizer implements GroovyCodeVisitor {
-    private int iota=0;
+    private static final AtomicLong iota = new AtomicLong();
```

Comment:
No, this class is loaded once in a given Jenkins session. There will be at least one _instance_ per `WorkflowRun` (the problem was that it might be more than one).